Monika After Story/game/script-apologies.rpy

```diff
@@ -0,0 +1,250 @@
+
+init -10 python in mas_apology:
+    apology_db = {}
+
+
+#going to need this post ev_handler init
+init 20 python:
+    def mas_checkApologies():
+        #Let's not do extra work
+        if persistent._mas_apology_time_db == {}:
+            return
+
+        #Calculate the current total playtime to compare...
+        current_total_playtime = persistent.sessions['total_playtime'] + (datetime.datetime.now() - persistent.sessions['current_session_start'])
+
+        #Iter thru the stuffs in the apology time tb
+        for ev, td in persistent._mas_apology_time_db.items():
+            if current_total_playtime >= td[0] or datetime.date.today() >= td[1]:
+                #Pop the ev_label from the time db and lock the event label. You just lost your chance
+                store.mas_hideEVL(ev,'APL',lock=True)
+                del persistent._mas_apology_time_db[ev]
+
+        return
+
+
+init 5 python:
+   addEvent(
+       Event(
+           persistent.event_database,
+           eventlabel='monika_playerapologizes',
+           prompt="I want to apologize...",
+           category=['you'],
+           pool=True,
+           unlocked=True
+        )
+    )
+
+label monika_playerapologizes:
+
+    #Firstly, let's check if there's an apology reason for the prompt
+    #NOTE: When adding more apology reasons, add a reason the player would say sorry for here (corresponding to the same #as the apology reason)
+    $ player_apology_reasons = {
+        0: "something else.", #since we shouldn't actually be able to get this, we use this as our fallback
+        1: "saying I wanted to break up.",
+        2: "joking about having another girlfriend.",
+        3: "calling you a murderer.",
+        4: "closing the game on you.",
+        5: "entering your room without knocking.",
+        6: "missing Christmas.",
+        7: "forgetting your birthday.",
+        8: "not spending time with you on your birthday.",
+        9: "the game crashing.",
+        10: "the game crashing." #easiest way to handle this w/o overrides
+    }
+
+    #Set the prompt for this...
+    if persistent._mas_apology_time_db != {}:
+        #If there's a non-generic apology reason pending we use "for something else."
+        $ mas_getEV('mas_apology_generic').prompt = "...for " + player_apology_reasons.get(mas_apology_reason,player_apology_reasons[0])
+    else:
+        #Otherwise we just use "for something."
+        $ mas_getEV('mas_apology_generic').prompt = "...for " + player_apology_reasons.get(mas_apology_reason,"something.")
+
+    #Then we delete this since we're not going to need it again until we come back here, where it's created again.
+    #No need to store excess memory
+    $ del player_apology_reasons
+
+    #Now we run through our apology db and find what's unlocked
+    python:
+        apologylist = [
+            (mas_getEV(ev).prompt,ev,False,False)
+            for ev in store.mas_apology.apology_db
+            if mas_getEV(ev).unlocked
+        ]
```

Comment:
so making the Event database means that this can be made like:
```python
apologylist = [
    (ev.prompt, ev, False, False)
    for ev_label, ev in store.mas_apology.apology_db.iteritems()
    if ev.unlocked
]
```